cmd/charm/charmcmd/cmd.go

```diff
@@ -203,30 +203,32 @@ func ussoTokenPath() string {
 	return osenv.JujuXDGDataHomePath("store-usso-token")
 }
 
-// errorMessage translates err into a user understandable error message,
-// and outputs the message ctxt.Stderr. If a message is output then
-// cmd.ErrSilent is returned, othewise err is returned unchanged.
-func errorMessage(ctxt *cmd.Context, err error) error {
+// translateError translates err into a new error with a more
+// understandable error message. If err is not translated then it will be
+// returned unchanged.
+func translateError(err error) error {
 	if err == nil {
 		return err
 	}
 	cause := errgo.Cause(err)
 	switch {
 	case httpbakery.IsInteractionError(cause):
-		return interactionErrorMessage(ctxt, cause.(*httpbakery.InteractionError))
+		return translateInteractionError(cause.(*httpbakery.InteractionError))
```

Comment:
How about:

    err := translateInteractionError(cause.(*http.InteractionError))
    return errgo.Notef(err, "login failed")

then we can get rid of a lot of indentation in the function below by returning early.